packages/commonwealth/client/scripts/views/components/component_kit/polls/index.tsx

```diff
@@ -0,0 +1,243 @@
+                      <div className="multi-select-votes">
+                        {voteInformation.map((option) => (
+                          <CWCheckbox
+                            checked={false}
+                            label={option.label}
+                            onchange={() => {
+                              // TODO: Build this out when multiple vote options are introduced.
+                              // Something like: this.selectedOptions.push(option.value);
+                              console.log('A vote for multiple options');
+                            }}
+                          />
+                        ))}
+                      </div>
+                    ) : (
+                      <div className="single-select-votes">
+                        {voteInformation.map((option) => (
+                          <CWRadioButton
+                            checked={
+                              this.selectedOptions.length > 0 &&
+                              option.value === this.selectedOptions[0]
+                            }
+                            groupName="votes"
+                            onchange={() => {
+                              this.selectedOptions[0] = option.value;
+                            }}
+                            label={option.label}
+                            value={option.value}
+                          />
+                        ))}
+                      </div>
+                    )}
+                  </div>
+                  <div className="cast-vote-section">
+                    <CWButton
+                      label="Vote"
+                      buttonType="mini"
+                      disabled={disableVoteButton}
+                      onclick={async () => {
+                        if (
+                          multiSelect ||
+                          this.selectedOptions[0] === votedFor ||
+                          this.selectedOptions.length === 0
+                        ) {
+                          // TODO: Build this out when multiple vote options are introduced.
+                          return;
+                        }
+                        if (pollType === PollType.Offchain) {
+                          await onVoteCast(
+                            this.selectedOptions[0],
+                            this.selectedOptions.length === 0,
+                            () => {
+                              if (!votedFor) {
+                                this.totalVoteCount += 1;
+                              }
+                              this.voteDirectionString =
+                                buildVoteDirectionString(
+                                  this.selectedOptions[0]
+                                );
+                              this.hasVoted = true;
+                            }
+                          );
+                        } else if (pollType === PollType.Snapshot) {
```

Comment:
I'm not sure I like that this is one component with both "offchain" and "snapshot" as polltypes. I would like the snapshot stuff factored out into its own component, SnapshotPollCard, to disambiguate the logic here. Better to keep it simple for clarity; the if/else statement for views based on poll type is bad practice (though seen in our app historically). 
```
/// best to keep it easier to read and edit like:
if snapshot, 
<SnapshotPollCard />
if poll,
<PollCard /> 
```